src/json_string_utilities.F90

```diff
@@ -134,6 +135,45 @@ subroutine real_to_string(rval,real_fmt,compact_real,str)
     end if
 
     end subroutine real_to_string
+
+!*****************************************************************************************
+!> author: Jacob Williams
+!  date: 12/4/2013
+!
+!  Convert a real value to a string.
+!
+!# Modified
+!  * Izaak Beekman : 02/24/2015 : added the compact option.
+!  * Jacob Williams : 10/27/2015 : added the star option.
+
+    subroutine real32_to_string(rval,real_fmt,compact_real,str)
+
+    implicit none
+
+    real(RK32),intent(in)                  :: rval         !! real value.
+    character(kind=CDK,len=*),intent(in) :: real_fmt     !! format for real numbers
+    logical(LK),intent(in)               :: compact_real !! compact the string so that it is
+                                                         !! displayed with fewer characters
+    character(kind=CK,len=*),intent(out) :: str          !! `rval` converted to a string.
+
+    integer(IK) :: istat
+
+    if (real_fmt==star) then
+        write(str,fmt=*,iostat=istat) rval
+    else
+        write(str,fmt=real_fmt,iostat=istat) rval
+    end if
+
+    if (istat==0) then
+        !in this case, the default string will be compacted,
+        ! so that the same value is displayed with fewer characters.
+        if (compact_real) call compact_real_string(str)
+    else
+        str = repeat(star,len(str))
+    end if
+
+    end subroutine real32_to_string
```

Comment:
Is this subroutine necessary? Couldn't you just create a wrapper for the real_to_string that handles doubles and just cast the 32 bit real as a 64bit real before passing it? (You could use different default edit descriptors like you do here and just pass them in to the original to_string subroutine.)